ci/semver.sh

```diff
@@ -0,0 +1,317 @@
+#!/usr/bin/env bash
+
+# ATTRIBUTION:
+# This is from https://github.com/fsaintjacques/semver-tool/blob/3.1.0/src/semver
+# Which has a GPL 3.0 license: https://choosealicense.com/licenses/gpl-3.0/
+
+
+set -o errexit -o nounset -o pipefail
+
+NAT='0|[1-9][0-9]*'
+ALPHANUM='[0-9]*[A-Za-z-][0-9A-Za-z-]*'
+IDENT="$NAT|$ALPHANUM"
+FIELD='[0-9A-Za-z-]+'
+
+SEMVER_REGEX="\
+^[vV]?\
+($NAT)\\.($NAT)\\.($NAT)\
+(\\-(${IDENT})(\\.(${IDENT}))*)?\
+(\\+${FIELD}(\\.${FIELD})*)?$"
+
+PROG=semver
+PROG_VERSION="3.1.0"
+
```

Comment:
@jeffmaher I would love to talk more about this particular file when you have time and why REGEX is used, just curious!

Reply:
@SMakaiTakori : Happy to chat about it! With that said, this is entirely taken from someone else and just worked, with the original repo noted at the top of the file.